### ISSUE TITLE
Why is author derived from latest commit?

### ISSUE THREAD

**Issue by blozano-tt** [fpicalausa/remove-stale-branches#33]
When person A creates a branch, but person B is the last person to commit to it.

Person B will be tagged/notified when the branch is stale.

Why?

**Comment by fpicalausa** [comment]
My understanding, there is no exact definition of a branch author. It could be:
- the author of first commit since the branch was derived from master, 
- the author first commit that is not on any other branch
- the authors of any commit not on master (except perhaps for merge commits or cherry picks)
- the author off any PR attached to the branch
- the author of the commit pointed to by the branch (i.e. last person to commit)
- etc.

Here, the goal is to alert someone that a branch they worked on is going to be dropped. Taking the commit currently pointed at is just a very simple approximation that happens to work well if you have contributors working on separate branches. With mixed contributors it may not work that well.

I assume your question stems from a very practical issue you faced. What would be your suggestion on the best approach?

**Comment by blozano-tt** [comment]
To answer in your terms. The author of the first commit in the branch makes more sense to me.

**Comment by fpicalausa** [comment]
I agree that it is a pitfall of the current approach. Here are two other scenarios I've encountered:
- the branch creator left the project; the latest contributor is the current interested person (in this case, the current approach works)
- the branch creator is an automated process that never commits (branch cut-off in case of weekly train releases)

Perhaps it'd be best to have a configurable strategy and let the action user decide on what works best with how they work with github. Thoughts?

**Comment by blozano-tt** [comment]
Yes that makes perfect sense.
Notify first or last author would be a great setting.

**Comment by HKBUSeanTYH** [comment]
Hi @fpicalausa, I think this is some what related to the notification mechanism, so I m adding a comment instead of creating a new issue. 

Currently it only notifies a singular person (whoever made the last commit) and if the person doesn’t exist, it notifies the default recipient (if configured). 

Is there any chance it could be configured to notify the default recipient always as well as whoever is the l first/last author?
